test/evergreen/setup_spawn_host.sh

```diff
@@ -23,6 +23,21 @@ if [[ -n $WT_ARCHIVE ]]; then
     tar --wildcards -xzf $WT_ARCHIVE
 fi
 
+# Setup the gdb environment if there are core dumps present in the artefacts.
+GDB_CORE_DUMP=$(ls ${HOME}/wiredtiger/cmake_build/*.core | head -n 1 2>/dev/null)
+if [[ -n $GDB_CORE_DUMP ]]; then
+    # Read the wiredtiger shared library and find the old source directory using readelf and fix up
+    # the expected path to the correct path on the new machine.
+    LIB_PATH=$(ls ${HOME}/wiredtiger/cmake_build/*.so* | head -n 1 2>/dev/null)
+    OLD_WT_PATH=$(readelf --debug-dump=info ${LIB_PATH} | grep "wiredtiger" | head -n 1 | sed -e 's/wiredtiger.*/wiredtiger/' -e 's/[^\/]*//')
+    cat >> ~/.gdbinit << EOF
+set solib-search-path ${HOME}/wiredtiger/cmake_build/lang/python:${HOME}/wiredtiger/cmake_build:${HOME}/wiredtiger/TCMALLOC_LIB/lib
+set substitute-path ${OLD_WT_PATH} ${HOME}/wiredtiger
```

Comment:
What does this do?

Reply:
@ajmorton explained to me, all good!

Reply:
If it raises confusion there's no harm in adding a comment. @jiechenbo can we explain the purpose of `set substitute-path` here and why we need it?

Reply:
done! [275063e](https://github.com/wiredtiger/wiredtiger/pull/9556/commits/275063e6f212a378f66a4a0e54fdb786f7b8403d)

Reply:
Thanks, could we add a bit more context here? Something along the lines of 
`gdb debug symbols include a path to the source code we compiled from, however since we compiled on a different machine and copied the artefacts to this spawn host the file paths have changed. substitute-path tells gdb the new location of our source code`

Reply:
thank you !